compiler/testData/codegen/box/parametersMetadata/suspendFunction.kt

```diff
@@ -0,0 +1,25 @@
+// IGNORE_BACKEND_FIR: JVM_IR
+// SKIP_JDK6
+// TARGET_BACKEND: JVM
+// WITH_RUNTIME
+// FULL_JDK
+// KOTLIN_CONFIGURATION_FLAGS: +JVM.PARAMETERS_METADATA
+// COMMON_COROUTINES_TEST
+import COROUTINES_PACKAGE.*
+import COROUTINES_PACKAGE.intrinsics.*
+
+class A() {
+    suspend fun test(OK: String) {
+
+    }
+}
+
+fun box(): String {
+    val clazz = A::class.java
+    val method = clazz.getDeclaredMethod("test", String::class.java, Continuation::class.java)
+    val parameters = method.getParameters()
+
+    if (parameters[0].modifiers != 0) return "wrong modifier on value parameter: ${parameters[0].modifiers}"
+    if (parameters[1].modifiers != 0) return "wrong modifier on Continuation parameter: ${parameters[1].modifiers}"
```

Comment:
I only checked the modifier and not the name because it's `$completion` in IR and `continuation` in the old backend/non-IR. In the old backend, the name in the parameter descriptor (used in the metadata) is `continuation` yet in the local variable table it's `$completion`. In IR, it's consistently `$completion`.

Should we change the name in the parameter descriptor for the old backend?

Reply:
cc @ilmirus

Reply:
@ilmirus @max-kammerer ping?

Reply:
@ilmirus @max-kammerer Gentle ping again!

Reply:
There is an issue with renaming continuation parameter in old BE with inliner, where it cannot inline non-suspend lambdas properly, will investigate and notify you, when this is done. I totally agree, that the parameter should be renamed to `$competion`, since it is `$completion` in LVT.
For now, I accept the PR.